### PR TITLE
add UTs for runtime handler

### LINKED ISSUE #69: [BUG] lack of unit, integration and functional test
**What you expected to happen**:
Right now, there are only few tests for each function. More tests for this project are needed.

image

### PR DESCRIPTION
**Pull request by saintube**
Signed-off-by: saintube <saintube@foxmail.com>

### Ⅰ. Describe what this PR does

add unit tests for package `pkg/runtime`

### Ⅱ. Does this pull request fix one issue?

fixes #69 

### Ⅲ. Describe how to verify it

run UTs and see the rise of code coverage

### Ⅳ. Special notes for reviews

none

### CODE REVIEW

File: pkg/runtime/handler/docker_runtime.go [hunk 1]
```diff
@@ -19,66 +19,87 @@ package handler
...
 }
 
-func createDockerClient(endpoint string) (*client.Client, error) {
-	ep := strings.TrimPrefix(endpoint, "unix://")
-	if _, err := os.Stat(ep); err != nil {
+func createDockerClient(httpClient *http.Client, endPoint string) (*dclient.Client, error) {
+	ep := strings.TrimPrefix(endPoint, "unix://")
+
+	_, err := os.Stat(ep)
+	if err != nil {
```
Comment (jasonliu747) [inline pkg/runtime/handler/docker_runtime.go:72]: make it one line `if _, err := os.Stat(ep); err != nil`

File: pkg/runtime/handler/docker_runtime.go [hunk 2]
```diff
@@ -19,66 +19,87 @@ package handler
...
-	return client.NewClientWithOpts(client.WithHost(endpoint))
+
+	return dclient.NewClientWithOpts(dclient.WithHost(endPoint), dclient.WithHTTPClient(httpClient))
 }
 
 func (d *DockerRuntimeHandler) StopContainer(containerID string, timeout int64) error {
-	if containerID == "" {
-		return fmt.Errorf("container ID cannot be empty")
+	if d == nil || d.dockerClient == nil {
+		return fmt.Errorf("stop container fail! docker client is nil!containerId=%v", containerID)
```
Comment (jasonliu747) [inline pkg/runtime/handler/docker_runtime.go:81]: add space between `!` and `containerId`

File: pkg/runtime/handler/docker_runtime.go [hunk 3]
```diff
@@ -19,66 +19,87 @@ package handler
...
 	}
 
-	if d == nil || d.dockerClient == nil {
-		return fmt.Errorf("failed to stop container %v, docker client is nil", containerID)
+	if containerID == "" {
+		return fmt.Errorf("ID cannot be empty")
```
Comment (jasonliu747) [inline pkg/runtime/handler/docker_runtime.go:85]: container ID

File: pkg/runtime/handler/docker_runtime.go [hunk 4]
```diff
@@ -19,66 +19,87 @@ package handler
...
 	stopTimeout := time.Duration(timeout) * time.Second
+
 	return d.dockerClient.ContainerStop(ctx, containerID, &stopTimeout)
 }
 
 func (d *DockerRuntimeHandler) UpdateContainerResources(containerID string, opts UpdateOptions) error {
-	if containerID == "" {
-		return fmt.Errorf("container ID cannot be empty")
+	if d == nil || d.dockerClient == nil {
+		return fmt.Errorf("UpdateContainerResources fail! docker client is nil!containerId=%v", containerID)
```
Comment (jasonliu747) [inline pkg/runtime/handler/docker_runtime.go:98]: add space between `!` and `containerId`

File: pkg/runtime/handler/docker_runtime.go [hunk 5]
```diff
@@ -19,66 +19,87 @@ package handler
...
 	}
 
-	if d == nil || d.dockerClient == nil {
-		return fmt.Errorf("failed to stop container %v, docker client is nil", containerID)
+	if containerID == "" {
+		return fmt.Errorf("ID cannot be empty")
```
Comment (jasonliu747) [inline pkg/runtime/handler/docker_runtime.go:102]: container ID

File: pkg/runtime/handler/docker_runtime.go [hunk 6]
```diff
@@ -94,7 +115,15 @@ func (d *DockerRuntimeHandler) UpdateContainerResources(containerID string, opts
 
 	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
 	defer cancel()
-	_, err := d.dockerClient.ContainerUpdate(ctx, containerID, updateConfig)
 
+	_, err := d.dockerClient.ContainerUpdate(ctx, containerID, updateConfig)
 	return err
 }
+
+func (d *DockerRuntimeHandler) Info() (string, error) {
```
Comment (jasonliu747) [inline pkg/runtime/handler/docker_runtime.go:123]: Looks like `Info()` is not used

File: pkg/runtime/handler/docker_runtime.go (hunk 1 above)
Comment (saintube) [reply pkg/runtime/handler/docker_runtime.go]: done

File: pkg/runtime/handler/docker_runtime.go (hunk 6 above)
Comment (saintube) [reply pkg/runtime/handler/docker_runtime.go]: fixed

**Review by jasonliu747 — COMMENTED** [review]
/lgtm
other PTAL

**Pull request merged**